projects/schematics/src/migrations/2_0/ssr/ssr_spec.ts

```diff
@@ -0,0 +1,211 @@
+import {
+  SchematicTestRunner,
+  UnitTestTree,
+} from '@angular-devkit/schematics/testing';
+import { runMigration } from '../../../shared/utils/test-utils';
+
+describe('ssr', () => {
+  let appTree: UnitTestTree;
+  let appSchematicRunner: SchematicTestRunner;
+  const workspaceOptions: any = {
+    name: 'workspace',
+    version: '9.0.0',
+  };
+
+  const appOptions: any = {
+    name: 'schematics-test',
+    inlineStyle: false,
+    inlineTemplate: false,
+    routing: false,
+    style: 'scss',
+    skipTests: false,
+    projectRoot: '',
+  };
+
+  beforeEach(async () => {
+    appSchematicRunner = new SchematicTestRunner(
+      'test',
+      require.resolve('../../migrations.json')
+    );
+
+    appTree = await appSchematicRunner
+      .runExternalSchematicAsync(
+        '@schematics/angular',
+        'workspace',
+        workspaceOptions
+      )
+      .toPromise();
+    appTree = await appSchematicRunner
+      .runExternalSchematicAsync(
+        '@schematics/angular',
+        'application',
+        appOptions,
+        appTree
+      )
+      .toPromise();
+
+    appTree.create('/server.ts', 'server content');
+    appTree.create('/webpack.server.config.js', 'webpack config content');
+
+    appTree.create(
+      '/src/main.server.ts',
+      `
+    import { enableProdMode } from '@angular/core';
+    import { environment } from './environments/environment';
+    if (environment.production) {
+      enableProdMode();
+    }
+    export { AppServerModule } from './app/app.server.module';
+    export { renderModule, renderModuleFactory } from '@angular/platform-server';
+    export { ngExpressEngine } from '@nguniversal/express-engine';
+    export { provideModuleMap } from "@nguniversal/module-map-ngfactory-loader";
+    `
+    );
+
+    appTree.create(
+      '/src/app/app.server.module.ts',
+      `
+    import { NgModule } from '@angular/core';
```

Comment:
For consistency with other test files, maybe we can move this to a const?

Reply:
Good point, thanks!